### DISCUSSION TITLE
Can convert the results of mcmc?

### DISCUSSION THREAD

**Discussion by d11122** [phoebe-project/phoebe2#524] [Questions]
Hi, I have a question about mcmc samples.
For example, I performed a mcmc, I performed parameters with 'incl', 'sma','r1','t2' and so on。
now I can  get the value and sigmas of M1 (see figure1)
![M1](https://user-images.githubusercontent.com/83321305/133090882-0421d571-ab88-4f18-adf7-e5fde6e6fc3f.png)
but my question is how to draw the corner figure with M1.
I can't find samples of M1 to plot corner figure.
Does it have to be involved in the mcmc calculation at the beginning?
thanks you!

**Comment by kecnry** [comment]
You should be able to create a corner plot with `b.plot(style='corner')` (you will also need to pass the `solution` tag if you have more than one).  If you want to plot a subset of the sampled parameters, you can either set the `adopt_parameters` parameter in the solution, or pass it as a keyword argument to plot.  If you want to plot parameters that were not sampled but are affected by existing constraints, you can do so as well by passing `parameters` as a list.

If you want to access the arrays and plot manually, see the parameter in the solution tagged with `qualifier='sampled'` or access them with [phoebe.helpers.process_mcmc_chains_from_solution](http://phoebe-project.org/docs/2.3/api/phoebe.helpers.process_mcmc_chains_from_solution).

You might also find these tutorials/docs helpful:
* [emcee tutorial](http://phoebe-project.org/docs/2.3/tutorials/emce) (scroll down to plotting at the bottom)
* [distribution propagation](http://phoebe-project.org/docs/2.3/tutorials/distribution_propagation)
* [plot API docs](http://phoebe-project.org/docs/2.3/api/phoebe.parameters.ParameterSet.plot)

Hope this helps!  Let us know if you run into any problems or questions.

**Reply by d11122** [reply]
Hi kecnry, thank your response, I can plot the corner figure with not sampled parameters by using phoebe `distribution propagation`.
For example The figure below shows ,the `r2/r1.`
![下载](https://user-images.githubusercontent.com/83321305/133184085-4dbdf49a-a643-4869-9738-a4ab78e38aad.png)
But I still can't find where the sample data for r2/r1 is saved? If I want to manually draw parameters that are not sampled but are affected by existing constraints .
I hope you will answer again.
thanks!

**Reply by kecnry** [reply]
yes, you can pass those same arguments to [b.sample_distribution_collection](http://phoebe-project.org/docs/2.3/api/phoebe.frontend.bundle.Bundle.sample_distribution_collection) and I think that should give you what you want (note that you'll likely want to pass `sample_size` and that depending on the cost of the constraint and the passed `sample_size`, this can get somewhat expensive).

**Comment by d11122** [comment]
Hi, kecnry, Thanks a lot, I have achieved the result I wanted.
